### PR TITLE
Backup: support TLS for br component

### PR DESCRIPTION
**Pull request by shuijing198799**
### What problem does this PR solve? 
use BR as the backup tool in an TLS cluster
### What is changed and how does it work?
use client certification for BR communicate with TiKV and PD
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
use operator and backup manager with this PR
```
helm install --values values.yaml --name operator --namespace tidb-admin . --set-string -operatorImage=hub.pingcap.net/yinliang/pingcap/tidb-operator:latest,tidbBackupManagerImage=hub.pingcap.net/yinliang/pingcap/tidb-backup-manager:latest
```

create backup and restore cluster
```
helm install --values values.yaml --name cluster --namespace cluster . --set-string enableTLSCluster=true,tidb.image=pingcap/tidb:latest,tikv.image=pingcap/tikv:latest,pd.image=pingcap/pd:latest
helm install --values values.yaml --name cluster1 --namespace cluster . --set-string enableTLSCluster=true,tidb.image=pingcap/tidb:latest,tikv.image=pingcap/tikv:latest,pd.image=pingcap/pd:latest
```

insert some data to tidbcluster cluster

create secret of tidb endpoint for mysql client
```
kubectl create secret generic backup-secret -n <namespace> --from-literal=user=<user> --from-literal=password=<password>
```

create secret for aws accesskey and secretkey
```
 kubectl create secret generic s3-secret-aws --from-literal=secret_key=<secret_key> --from-literal=access_key=<access_key> --namespace=cluster
 ```

apply backup rbac 
```
---
kind: Role
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: tidb-backup-manager
  labels:
    app.kubernetes.io/component: tidb-backup-manager
rules:
- apiGroups: [""]
  resources: ["events"]
  verbs: ["*"]
- apiGroups: ["pingcap.com"]
  resources: ["backups", "restores"]
  verbs: ["get", "watch", "list", "update"]

---
kind: ServiceAccount
apiVersion: v1
metadata:
  name: tidb-backup-manager

---
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: tidb-backup-manager
  labels:
    app.kubernetes.io/component: tidb-backup-manager
subjects:
- kind: ServiceAccount
  name: tidb-backup-manager
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: tidb-backup-manager
```

use backup yaml to create backup CRD
```
apiVersion: pingcap.com/v1alpha1
kind: Backup
metadata:
  name: jony-backup-aws-202003021703
  namespace: cluster
spec:
  br:
    enableTLSClient: true
    cluster: cluster
    clusterNamespce: cluster
  from:
    host: cluster-tidb
    port: 4000
    user: root
    secretName: backup-secret
  s3:
    provider: aws
    region: us-west-2
    secretName: s3-secret-aws
    bucket: backup.jony3.us-west-2.tidbcloud.com
    prefix: "202003021703"
  storageClassName: local-storage
  storageSize: 10Gi

  ```

check backup files from aws console

use restore yaml to restore 
```
apiVersion: pingcap.com/v1alpha1
kind: Restore
metadata:
  name: jony-restore-aws-202003021703
  namespace: cluster
spec:
  br:
    enableTLSClient: true
    cluster: cluster1
    clusterNamespce: cluster
  to:
    host: cluster1-tidb
    port: 4000
    user: root
    secretName: backup-secret
  s3:
    provider: aws
    region: us-west-2
    secretName: s3-secret-aws
    bucket: backup.jony3.us-west-2.tidbcloud.com
    prefix: "202003021703"
  storageClassName: local-storage
  storageSize: 10Gi
```

check the cluster and cluster1 data, whether they are consistence

Code changes

 - Has Go code change

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 1]
```diff
@@ -786,18 +786,16 @@ type BackupSpec struct {
 // +k8s:openapi-gen=true
 // BRConfig contains config for BR
 type BRConfig struct {
-	// PDAddress is the PD address of the tidb cluster
-	PDAddress string `json:"pd"`
+	// Whether enable TLS in TiDBCluster
+	EnableTLSClient bool `json:"enableTLSClient,omitempty"`
+	// ClusterName of backup cluster
```
Comment (LinuxGit) [inline pkg/apis/pingcap/v1alpha1/types.go]: s/backup/`backup/restore`

Comment (shuijing198799) [reply pkg/apis/pingcap/v1alpha1/types.go]: done

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 2]
```diff
@@ -786,18 +786,16 @@ type BackupSpec struct {
 // +k8s:openapi-gen=true
 // BRConfig contains config for BR
 type BRConfig struct {
-	// PDAddress is the PD address of the tidb cluster
-	PDAddress string `json:"pd"`
+	// Whether enable TLS in TiDBCluster
+	EnableTLSClient bool `json:"enableTLSClient,omitempty"`
+	// ClusterName of backup/restore cluster
+	Cluster string `json:"cluster,omitempty"`
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/types.go]: This cannot be empty, right?

Comment (shuijing198799) [reply pkg/apis/pingcap/v1alpha1/types.go]: yes，I have verify it in the validate function

**Pull request merged**